README.md

```diff
@@ -1,2 +1,283 @@
+               <data android:scheme="msal<YOUR_CLIENT_ID>"
+                    android:host="auth" />
+          </intent-filter>
+     </activity>
+
+## 2. Getting started
+
+### 2.1 Register your android application
+
+Register your application by following the steps at [Register your app with the Azure AD v2.0 endpoint](https://developer.microsoft.com/en-us/graph/docs/concepts/auth_register_app_v2).
+
+### 2.2 Create an MSALAuthenticationProvider object
+
+An instance of the **GraphServiceClient** class handles building requests, sending them to the Microsoft Graph API, and processing the responses. To create a new instance of this class, you need to provide an instance of `MSALAuthenticationProvider`, which can authenticate requests to Microsoft Graph.
+
+```
+PublicClientApplication publicClientApplication = new PublicClientApplication(getApplicationContext(), "CLIENT_ID_OF_YOUR_APPLICATION");
+MSALAuthenticationProvider msalAuthenticationProvider = new MSALAuthenticationProvider(
+    getActivity(),
+    getApplication(),
+    publicClientApplication,
+    scopes);
+```
+
+### 1.5 Add Auth helper in your activity calling graph
+//Create an onActivityResult method
+/* Handles the redirect from the System Browser */
+@Override
+protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+     msalAuthenticationProvider.handleInteractiveRequestRedirect(requestCode, resultCode, data);
+}
+
+### 2.3 Get a GraphServiceClient object
+After you have set the correct application ID/client ID, you must get a **GraphServiceClient** object to make requests against the service. The SDK stores the account information for you, but when a user signs in for the first time, it invokes the UI to get the user's account information.
+
+```java
+IGraphServiceClient graphClient =
+  GraphServiceClient
+    .builder()
+    .authenticationProvider(msalAuthenticationProvider)
+    .buildClient();
+```
+
+## 3. Make requests against the service
+
+After you have a GraphServiceClient object, you can begin making calls against the service. The requests against the service look like our [REST API](https://developer.microsoft.com/en-us/graph/docs/concepts/overview).
+
+### 3.1 Get the user's drive
+
+To retrieve the user's drive:
+
+```java
+graphClient
+  .me()
+  .drive()
+  .buildRequest()
+  .get(new ICallback<Drive>() {
+     @Override
+     public void success(final Drive result) {
+        System.out.println("Found Drive " + result.id);
+     }
+     ...
+     // Handle failure case
+  });
+```
+
+### 4 Sample
+
+AndroidManifest.xml
+```
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="your.domain.name">
+
+    <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
+
+    <application
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
```

Comment:
why are you putting everything from manifest file in the readme, only put relevant part in the readme, remove the common part.

Reply:
removed sample from readme.